### PR TITLE
onik/deriv-com_pagespeed

### PR DESCRIPTION
**Pull request by onik-deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: deriv.com.conf [hunk 1]
```diff
@@ -102,4 +102,33 @@ server {
...
+    # and no extraneous headers get set.
+    location ~ "\.pagespeed\.([a-z]\.)?[a-z]{2}\.[^.]{10}\.[^.]+" {
+     add_header "" "";
+    }
+    location ~ "^/pagespeed_static/" { }
+    location ~ "^/ngx_pagespeed_beacon$" { }
+    #charset koi8-r;
+
+    #access_log  logs/host.access.log  main;
+
```
Comment (shudarshon-binary) [inline deriv.com.conf:120]: please commented lines that are related to configurations. you can move the module `on` to the upper side of the config as per the convention

File: deriv.com.conf [hunk 2]
```diff
@@ -102,4 +102,22 @@ server {
...
+    location ~ "\.pagespeed\.([a-z]\.)?[a-z]{2}\.[^.]{10}\.[^.]+" {
+     add_header "" "";
+    }
+    location ~ "^/pagespeed_static/" { }
+    location ~ "^/ngx_pagespeed_beacon$" { }
+
+    location / {
+        root   html;
+        index  index.html index.htm;
+    }
```
Comment (shudarshon-binary) [inline deriv.com.conf:116]: we dont need multiple roots, there is one declared already

File: deriv.com.conf [hunk 3]
```diff
@@ -102,4 +102,17 @@ server {
...
     }
+    pagespeed FileCachePath /var/ngx_pagespeed_cache;
+
+    location ~ "\.pagespeed\.([a-z]\.)?[a-z]{2}\.[^.]{10}\.[^.]+" {
+     add_header "" "";
+    }
+    location ~ "^/pagespeed_static/" { }
+    location ~ "^/ngx_pagespeed_beacon$" { }
+
+    error_page   500 502 503 504  /50x.html;
```
Comment (shudarshon-binary) [inline deriv.com.conf:113]: we already have error_page code declared. please remove the redundant one

File: deriv.com.conf [hunk 4]
```diff
@@ -102,4 +102,16 @@ server {
...
+
+    location ~ "\.pagespeed\.([a-z]\.)?[a-z]{2}\.[^.]{10}\.[^.]+" {
+     add_header "" "";
+    }
+    location ~ "^/pagespeed_static/" { }
+    location ~ "^/ngx_pagespeed_beacon$" { }
+
+    location = /50x.html {
+       root   html;
+    }
```
Comment (shudarshon-binary) [inline deriv.com.conf:115]: better to avoid this block as we have already declared error_page to 50x status codes

**Review by shudarshon-binary — CHANGES_REQUESTED** [review]

**Review by shudarshon-binary — APPROVED** [review]

**Review by yashim-deriv — APPROVED** [review]

**Review by habib-deriv — APPROVED** [review]

**Pull request merged**